### PR TITLE
Use anonymized return type for `Documents::iter`.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17542)

### CODE REVIEW

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -380,23 +380,8 @@ impl Documents {
...
-pub struct DocumentsIter<'a> {
-    iter: hash_map::Iter<'a, PipelineId, JS<Document>>,
-}
-
-impl<'a> Iterator for DocumentsIter<'a> {
-    type Item = (PipelineId, Root<Document>);
-
-    fn next(&mut self) -> Option<(PipelineId, Root<Document>)> {
-        self.iter.next().map(|(id, doc)| (*id, Root::from_ref(&**doc)))
+    pub fn iter<'a>(&'a self) -> impl Iterator<Item=(PipelineId, Root<Document>)> {
```
Comment (asajeffrey) [inline components/script/script_thread.rs]: Yes, this is so much nicer! impl return didn't exist when I wrote this originally.

**Comment by frewsxcv** [comment]
Thought I had compiled this locally but it looks it fails for me too. I looked into the compile error briefly but couldn't figure out what the issue is here and I'm short on time, so I'm just going to close this.

File: components/script/script_thread.rs (hunk 1 above)
Comment (cbrewster) [inline components/script/script_thread.rs]: @frewsxcv my guess is you need the `+ 'a` in `impl Iterator<Item=(PipelineId, Root<Document>)> + 'a`

Comment (frewsxcv) [inline components/script/script_thread.rs]: I got other unrelated lifetime errors after. They were fixable, but it didn't seem like the errors were necessary

**Comment by asajeffrey** [comment]
Oh that's annoying, the type of `map.iter()` includes a lifetime, so can't be returned in an impl.

**Pull request closed without merging**